src/EVault/modules/Token.sol

```diff
@@ -48,17 +48,32 @@ abstract contract TokenModule is IToken, Base, BalanceUtils {
     }
 
     /// @inheritdoc IERC20
-    function transfer(address to, uint256 amount) public virtual reentrantOK returns (bool) {
-        return transferFrom(address(0), to, amount);
+    function transfer(address to, uint256 amount) public virtual nonReentrant returns (bool) {
+        return transferFromInternal(address(0), to, amount);
     }
 
     /// @inheritdoc IToken
-    function transferFromMax(address from, address to) public virtual reentrantOK returns (bool) {
-        return transferFrom(from, to, vaultStorage.users[from].getBalance().toUint());
+    function transferFromMax(address from, address to) public virtual nonReentrant returns (bool) {
+        validateTransferFromAccount(from);
+        return transferFromInternal(from, to, vaultStorage.users[from].getBalance().toUint());
     }
 
     /// @inheritdoc IERC20
     function transferFrom(address from, address to, uint256 amount) public virtual nonReentrant returns (bool) {
+        validateTransferFromAccount(from);
+        return transferFromInternal(from, to, amount);
+    }
+
+    /// @inheritdoc IERC20
+    function approve(address spender, uint256 amount) public virtual nonReentrant returns (bool) {
+        address account = EVCAuthenticate();
+
+        setAllowance(account, spender, amount);
+
+        return true;
+    }
+
+    function transferFromInternal(address from, address to, uint256 amount) private returns (bool) {
         (, address account) = initOperation(OP_TRANSFER, from == address(0) ? CHECKACCOUNT_CALLER : from);
 
         if (from == address(0)) from = account;
```

Comment:
Hmmm so we still are using the `address(0)` hack, just paper over it by blocking these addresses?